.github/workflows/pull_request.yaml

```diff
@@ -54,6 +54,50 @@ jobs:
           verbose: true
           fail_ci_if_error: false
 
+  functional-tests:
+      runs-on: ubuntu-latest
+      timeout-minutes: 15
+      steps:
+        - uses: actions/checkout@v3
+
+        - uses: actions/setup-go@v3
+          with:
+            go-version-file: './go.mod'
+
+        - id: go-cache-paths
+          run: |
+            echo "::set-output name=go-build::$(go env GOCACHE)"
+            echo "::set-output name=go-mod::$(go env GOMODCACHE)"
+
+        # Cache go build cache, used to speedup go test
+        - uses: actions/cache@v3
+          with:
+            path: ${{ steps.go-cache-paths.outputs.go-build }}
+            key: ${{ runner.os }}-go-build-${{ hashFiles('**/go.sum') }}
+
+        # Cache go mod cache, used to speedup builds
+        - uses: actions/cache@v3
+          with:
+            path: ${{ steps.go-cache-paths.outputs.go-mod }}
+            key: ${{ runner.os }}-go-mod-${{ hashFiles('**/go.sum') }}
+
+        # build the openfga/openfga image for the functional tests
+        - uses: docker/build-push-action@v2
+          with:
+            file: Dockerfile
+            push: false
```

Comment:
Could you please document this so that it doesn't get randomly flipped?

Reply:
Added a comment to it